### ISSUE TITLE
Option hot_standby

### ISSUE THREAD

**Issue by kostja** [tarantool/tarantool#775]
The problem: it's very easy to start two instances of the same server in the same data directory.
Not so easy if tarantoolctl is used, but even with tarantoolctl one can make a typo and have two instances listening on the same port, making a case for misconfigured local hot standby.

Solution:

- by default, we will lock the WAL directory on start, and abort start if the directory is locked
- we will not re-try binding to the listen port indefinitely; instead, we will fail to start

If hot_standby (a new option) is on, we will ignore the missing lock on the wal directory, and boot the snapshot, and begin following the WAL directory and reading up xlogs (hot standby mode).
As soon as we get the lock, we'll bind (no bind retries, fail on first attempt) and enter the master mode.

In other words, we'll stop using the listen port as an indicator of hot_standby mode, will require an explicit option (hot_standby), and will enter hot standby only after acquiring a filesystem lock on WAL dir path.

**Comment by kostja** [comment]
@rybakit got affected by this

**Comment by mejedi** [comment]
So are we going to add a new parameter in box.cfg{} that defaults to false and in order for the several instances to work as a local hot standby ensemble all of them must enable it explicitly?

There is a drawback with another lock file: if the second instance 'consumed' a file and then crashed and got restarted there is no unlock file anymore.

Another approach would be for the second instance to either:

 * exit if failed to bind the port (default);

 * or open a connection with that particular port, confirm that hot backup is enabled in the first instance, and wait for the connection to close. Once we get connection closed event it means that the first instance has crashed. A bonus: instant reaction.

Caveat: what if the first instance was restarted fast enough before the hot backup instance got a chance to claim the port?

**Comment by rtsisyk** [comment]
Discussed with @kostja => remove hot standby because it doesn't work on vinyl

**Comment by rtsisyk** [comment]
I suggest to keep hot standby as is in 1.7.x for memtx-only instances and disable it if instance has vinyl spaces.
